src/Access/EnabledSettings.h

```diff
@@ -40,17 +41,23 @@ class EnabledSettings
     /// and the roles passed in the constructor.
     std::shared_ptr<const SettingsConstraints> getConstraints() const;
 
+    const std::vector<UUID> & getAppliedProfiles() const;
```

Comment:
The function should not return `const std::vector<UUID> &`. It should return either `std::shared_ptr<const std::vector<UUID>>` or `std::vector<UUID>` instead. Because if it returns `const std::vector<UUID> &` it can be read outside the internal lock and be modified in `setSettingsAndConstraintsAndProfiles()` at the same time.

Reply:
Done